### PR TITLE
Update README.md for New Era

### PR DESCRIPTION
**Pull request by oxinabox**
- Remove WIP stuff, we are looking solid
 - Chain Goals to Features
    - thus remove "First class complex support" as it sure doesn't feel first class right now (still a goal but not a feature right now)
     - emphisize extensiability

- Explain relationship with ChainRulesCore

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,22 +3,24 @@
...
+and has all the rules for the julia standard library.
 
-Here are some of the basic goals for the package:
 
-- First-class support for complex differentiation via Wirtinger derivatives.
+Here are some of the core features of the package:
 
 - Mixed-mode composability without being coupled to a specific AD implementation.
-
+- Extensible rules: package authors can add rules (and thus AD support) to the functions in their packages in their packages. They don't need to edit the ChainRules repo.
```
Comment (nickrobinson251) [inline README.md]:
```suggestion
- Extensible rules: package authors can add rules (and thus AD support) to the functions in their packages, without needing to make a PR to ChainRules.jl .
```

File: README.md [hunk 2]
```diff
@@ -3,22 +3,24 @@
...
+- Control-inverted design: rule authors can fully specify derivatives in a concise manner while naturally allowing the caller to compute only what they need.
```
Comment (nickrobinson251) [inline README.md]: what does this mean?

File: README.md [hunk 3]
```diff
@@ -3,22 +3,24 @@
...
-**Docs:* 
+**Docs:**
 [![](https://img.shields.io/badge/docs-master-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/dev)
 [![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/stable)
 
 The ChainRules package provides a variety of common utilities that can be used by downstream automatic differentiation (AD) tools to define and execute forward-, reverse-, and mixed-mode primitives.
 
-This package is a WIP; the framework is essentially there, but there are a bunch of TODOs, virtually no tests, etc. PRs welcome! Documentation is incoming, which should help if you'd like to contribute.
+The Core logic of ChainRules is implemented in [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl).
+If adding ChainRules support to your package (defining `rrule`s or `ffules` you only need to depend on that very light-weight package.
```
Comment (nickrobinson251) [inline README.md]:
```suggestion
To add ChainRules support to your package, by defining new `rrule`s or `ffrules`, you only need to depend on the very light-weight package ChainRulesCore.jl.
```

File: README.md [hunk 4]
```diff
@@ -3,22 +3,24 @@
...
+This repo: ChainRules.jl is what people actually use, it reexports all the ChainRules functionality,
```
Comment (nickrobinson251) [inline README.md]: can we make these full sentences? e.g.
```suggestion
This repository contains ChainRules.jl, which is what people actually use directly. 
ChainRules reexports all the ChainRulesCore functionality, and has all the rules for the Julia standard library.
```
and one sentence per line makes diffs in follow-up PRs much nicer

File: README.md [hunk 5]
```diff
@@ -3,22 +3,24 @@
...
+and has all the rules for the julia standard library.
```
Comment (nickrobinson251) [inline README.md]:
```suggestion
```

File: README.md (hunk 2 above)
Comment (nickrobinson251) [reply README.md]: actually i think i know what it means, but it's taken me a couple readings to get there...

How about changing the last bit to something like this (still not totally happy with it)

```suggestion
- Control-inverted design: rule authors can fully specify derivatives in a concise manner that supports computation efficiencies, so we will only compute as much as the user requests.
```

File: README.md (hunk 1 above)
Comment (oxinabox) [reply README.md]: idk

File: README.md (hunk 5 above)
Comment (oxinabox) [reply README.md]: we have to say something like that.

Comment (nickrobinson251) [reply README.md]: this is to change to 1 sentence per line.

the content is included in the new full sentence in the suggestion one up (at https://github.com/JuliaDiff/ChainRules.jl/pull/112#discussion_r328636209)

File: README.md (hunk 4 above)
Comment (nickrobinson251) [reply README.md]: here @oxinabox

File: README.md (hunk 1 above)
Comment (nickrobinson251) [reply README.md]: at least change it to remove the repeated words?

File: README.md [hunk 6]
```diff
@@ -3,22 +3,25 @@
...
 
-**Docs:* 
+**Docs:**
 [![](https://img.shields.io/badge/docs-master-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/dev)
 [![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaDiff.github.io/ChainRules.jl/stable)
 
 The ChainRules package provides a variety of common utilities that can be used by downstream automatic differentiation (AD) tools to define and execute forward-, reverse-, and mixed-mode primitives.
 
-This package is a WIP; the framework is essentially there, but there are a bunch of TODOs, virtually no tests, etc. PRs welcome! Documentation is incoming, which should help if you'd like to contribute.
+The Core logic of ChainRules is implemented in [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl).
```
Comment (oxinabox) [inline README.md]:
```suggestion
The core logic of ChainRules is implemented in [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl).
```

File: README.md [hunk 7]
```diff
@@ -3,22 +3,25 @@
...
+To add ChainRules support to your package, by defining new `rrule`s or `ffrules`, you only need to depend on the very light-weight package ChainRulesCore.jl.
```
Comment (oxinabox) [inline README.md]:
```suggestion
To add ChainRules support to your package, by defining new `rrule`s or `frules`, you only need to depend on the very light-weight package ChainRulesCore.jl.
```

**Pull request merged**